internal/pkg/fleetapi/acker/lazy/lazy_acker.go

```diff
@@ -6,55 +6,99 @@ package lazy
 
 import (
 	"context"
+	"net/http"
 
 	"github.com/elastic/elastic-agent/internal/pkg/core/logger"
 	"github.com/elastic/elastic-agent/internal/pkg/fleetapi"
 )
 
 type batchAcker interface {
-	AckBatch(ctx context.Context, actions []fleetapi.Action) error
+	AckBatch(ctx context.Context, actions []fleetapi.Action) (*fleetapi.AckResponse, error)
 }
 
-type ackForcer interface {
-	ForceAck()
+type retrier interface {
+	Enqueue([]fleetapi.Action)
 }
 
 // Acker is a lazy acker which performs HTTP communication on commit.
 type Acker struct {
-	log   *logger.Logger
-	acker batchAcker
-	queue []fleetapi.Action
+	log     *logger.Logger
+	acker   batchAcker
+	queue   []fleetapi.Action
+	retrier retrier
 }
 
+// Option Acker option function
+type Option func(f *Acker)
+
 // NewAcker creates a new lazy acker.
-func NewAcker(baseAcker batchAcker, log *logger.Logger) *Acker {
-	return &Acker{
+func NewAcker(baseAcker batchAcker, log *logger.Logger, opts ...Option) *Acker {
+	f := &Acker{
 		acker: baseAcker,
 		queue: make([]fleetapi.Action, 0),
 		log:   log,
 	}
+
+	for _, opt := range opts {
+		opt(f)
+	}
+
+	return f
+}
+
+// WithRetrier option allows to specify the Retrier for acking
+func WithRetrier(r retrier) Option {
+	return func(f *Acker) {
+		f.retrier = r
+	}
 }
 
 // Ack acknowledges action.
 func (f *Acker) Ack(ctx context.Context, action fleetapi.Action) error {
 	f.enqueue(action)
-
-	if _, isAckForced := action.(ackForcer); isAckForced {
-		return f.Commit(ctx)
-	}
-
 	return nil
 }
 
 // Commit commits ack actions.
 func (f *Acker) Commit(ctx context.Context) error {
-	err := f.acker.AckBatch(ctx, f.queue)
+	if len(f.queue) == 0 {
+		return nil
+	}
+
+	actions := f.queue
+	f.queue = make([]fleetapi.Action, 0)
+
+	f.log.Debugf("lazy acker: ackbatch: %#v", actions)
+	resp, err := f.acker.AckBatch(ctx, actions)
+
+	// If request failed enqueue all actions with retrier if set
```

Comment:
```suggestion
	// If request failed enqueue all actions with retrier is set
```

Reply:
"enqueue all actions with retrier if it is set" probably better if "enqueue all actions with retrier if set" was not clear